crates/rust-analyzer/src/handlers/request.rs

```diff
@@ -1995,14 +1996,31 @@ fn run_rustfmt(
             cmd
         }
         RustfmtConfig::CustomCommand { command, args } => {
-            let mut cmd = process::Command::new(command);
+            let cmd = PathBuf::from(&command);
+            let workspace = CargoTargetSpec::for_file(&snap, file_id)?;
+            let mut cmd = match workspace {
+                Some(spec) => {
+                    // approach: if the command name contains a path seperator, join it with the workspace root.
+                    // however, if the path is absolute, joining will result in the absolute path being preserved.
+                    // as a fallback, rely on $PATH-based discovery.
+                    let cmd_path = if command.contains(std::path::MAIN_SEPARATOR_STR) {
+                        spec.workspace_root.join(cmd).into()
+                    } else {
+                        cmd
+                    };
```

Comment:
```suggestion
                    let cmd_path = if cfg!(windows) && command.contains(&[std::path::MAIN_SEPARATOR_STR, '/']) {
                        spec.workspace_root.join(cmd).into()
                    } else if command.contains(std::path::MAIN_SEPARATOR_STR) {
                        spec.workspace_root.join(cmd).into()
                    } else {
                        cmd
                    };
```
Since windows can use either separators.